docs/cmd/kn_service_create.md

```diff
@@ -61,6 +61,7 @@ kn service create NAME --image IMAGE [flags]
   -n, --namespace string         Specify the namespace to operate in.
       --no-lock-to-digest        do not keep the running image for the service constant when not explicitly specifying the image. (--no-lock-to-digest pulls the image tag afresh with each new revision)
   -p, --port int32               The port where application listens on.
+      --pull-secrets string      Image pull secrets to set. Empty image pull secrets will result to clear the pull secrets.
```

Comment:
This "Empty image pull secrets will result to clear the pull secrets" reads odd...  Perhaps: "Empty image pull secrets will clear the pull secrets field"

Reply:
Maybe: "No pull secret is used when no secret is provided as an argument"

btw, this is another example why we might to separate better the create and update help messages as for a `kn service create` an empty `--pull-secrets` doesn't make sense at all (and would be a no-op)

Reply:
>  this is another example why we might to separate better the create and update help messages as for a kn service create an empty --pull-secrets doesn't make sense at all (and would be a no-op)

+1, good point!

I kept the message in format as we've for ServiceAccount
```
      --service-account string   Service account name to set. Empty service account name will result to clear the service account.
```
The pull secrets are set per revision of the service, you can create a service specifying the pull secrets and a private image, then update service with another revision with a public image and clearing the pull secrets.
Also note, created a service with pull-secrets and private image, any subsequent updates to service without image and empty pull secrets (`""`) works fine too.

Reply:
I've updated the help text as below for `--pull-secrets` and `--service-account` flag.
```
Image pull secrets to set. Empty image pull secrets will the clear the pull secrets.

```
```
Service account name to set. Empty service account name will clear the service account.
```